godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/Vector2.kt

```diff
@@ -0,0 +1,311 @@
+
+        return ((p1 * 2.0) +
+            (-p0 + p2) * t +
+            (p0 * 2.0 - p1 * 5.0 + p2 * 4.0 - p3) * t2 +
+            (-p0 + p1 * 3.0 - p2 * 3.0 + p3) * t3) * 0.5
+    }
+
+    fun directionTo(other: Vector2): Vector2 {
+        val ret = Vector2(other.x - x, other.y - y)
+        ret.normalize()
+        return ret
+    }
+
+    fun distanceSquaredTo(other: Vector2): RealT {
+        return (x - other.x) * (x - other.x) + (y - other.y) * (y - other.y)
+    }
+
+    fun distanceTo(other: Vector2): RealT {
+        return sqrt((x - other.x) * (x - other.x) + (y - other.y) * (y - other.y))
+    }
+
+    fun dot(other: Vector2): RealT {
+        return x * other.x + y * other.y
+    }
+
+    fun floor(): Vector2 {
+        return Vector2(floor(x), floor(y))
+    }
+
+    fun isEqualApprox(other: Vector2): Boolean {
+        return isEqualApprox(other.x, x) && isEqualApprox(other.y, y)
+    }
+
+    fun isNormalized(): Boolean {
+        return isEqualApprox(this.length(), 1.0)
+    }
+
+    fun length(): RealT {
+        return sqrt(x * x + y * y)
+    }
+
+    fun lengthSquared(): RealT {
+        return x * x + y * y
+    }
+
+    fun linearInterpolate(v: Vector2, t: RealT): Vector2 {
+        val res: Vector2 = this
+        res.x += (t * (v.x - x))
+        res.y += (t * (v.y - y))
+        return res
+    }
+
+    fun moveToward(to: Vector2, delta: RealT): Vector2 {
+        val vd = to - this
+        val len = vd.length()
+        return if (len <= delta || len < CMP_EPSILON)
+            to
+        else
+            this + vd / len * delta
+    }
+
+    fun normalized(): Vector2 {
+        val v: Vector2 = this
+        v.normalize()
+        return v
+    }
+
+    internal fun normalize() {
+        val l: RealT = length()
+        if (l == 0.0) {
+            x = 0.0
+            y = 0.0
+        } else {
+            x /= l
+            y /= l
+        }
+    }
+
+    fun posmod(mod: RealT): Vector2 {
+        return Vector2(x.rem(mod), y.rem(mod))
+    }
+
+    fun posmodv(modv: Vector2): Vector2 {
+        return Vector2(x.rem(modv.x), y.rem(modv.y))
+    }
+
+    fun project(vec: Vector2): Vector2 {
+        val v1: Vector2 = vec
+        val v2: Vector2 = this
+        return v2 * (v1.dot(v2) / v2.dot(v2))
+    }
+
+    fun reflect(vec: Vector2): Vector2 {
+        return vec * this.dot(vec) * 2.0 - this
+    }
+
+    fun rotated(by: RealT): Vector2 {
+        var v = Vector2(0.0, 0.0)
+        v.rotate(this.angle() + by)
+        v *= length()
+        return v
+    }
+
+    internal fun rotate(radians: RealT) {
+        x = cos(radians)
+        y = sin(radians)
+    }
+
+    fun round(): Vector2 {
+        return Vector2(round(x), round(y))
+    }
+
+    fun sign(): Vector2 {
+        return Vector2(sign(x), sign(y))
+    }
+
+    fun slerp(b: Vector2, t: RealT): Vector2 {
+        if (!this.isNormalized() || !b.isNormalized())
+            Godot.printError("Vectors not normalized", "slerp()", "Vector2.kt", 228)
```

Comment:
Again i know you are not a fan of {} but IIRC kotlin convetion is also {} for single lines :-)